app/Bundle.php

```diff
@@ -53,9 +57,100 @@ class Bundle extends Model
     protected $dates = [
         'created_at',
         'updated_at',
-        'allocated_at'  // When it was handed out.
+        'disbursed_at'  // When it was handed out.
     ];
 
+    /**
+     * Refactored function that works out if we broke anything then adds vouchers.
+     *
+     * @param Collection $vouchers
+     * @param array $codes
+     * @param Bundle|null $bundle
+     * @return array
+     */
+    private function alterVouchers(Collection $vouchers, array $codes, Bundle $bundle = null)
+    {
+        $errors = [];
+        $badCodes = array_diff($vouchers->pluck("code")->toArray(), $codes);
+
+        if (!isEmpty($badCodes)) {
+            //Stop! report the bad codes!
+            $errors["codes"] = (array_key_exists("badCodes", $errors))
```

Comment:
Won't `$errors` always be `[]` here?

Reply:
hmm. probably an artefact of me refactoring this function out of the syncVouchers. Will amend.